scripts/git/prune_old_tags.sh

```diff
@@ -34,19 +45,32 @@ fi
 DELETE_TAGS_COUNT=$((TOTAL_TAGS_COUNT-KEEP_TAGS_COUNT))
 TAGS=($(git for-each-ref --sort=creatordate --count="$DELETE_TAGS_COUNT" --format '%(refname:short)' refs/tags))
 
+echo "$TOTAL_TAGS_COUNT tags available"
+echo "$DELETE_TAGS_COUNT tags will be deleted"
+
+if [[ "$DRY_RUN" != "DRY_RUN" ]]; then
+  read -p "Continue? (y/n) " -n 1 -r
+  echo
+  if [[ ! $REPLY =~ ^[Yy]$ ]]
+  then
+    echo "Exiting"
+    exit 0
+  fi
+fi
+
 # Filter out tags that were not created by snapshot releases.
 for TAG_INDEX in "${!TAGS[@]}"; do
   if [[ ! ${TAGS[TAG_INDEX]} =~ ^snapshot-* ]]; then
     unset 'TAGS[TAG_INDEX]'
   fi
 done
 
-if [[ $DRY_RUN -eq 1 ]]; then
-  echo "Dry run mode: these tags would be deleted"
+if [[ $DRY_RUN == "DRY_RUN" ]]; then
```

Comment:
This is a bit of a brittle way to read arguments, but it took me forever to figure out argument parsing in bash (and I hadn't when I wrote all our other scripts), so not going to press the point too much. But just think about cases like someone running `./prune_old_tags.sh 100 dryrun` or `./prune_old_tags.sh 100 --dry-run` `./prune_old_tags.sh 100 foobar`. As a simple way to make this less error prone and since you really only accept one argument so don't need general parsing, you could: make the check case and punctuation insensitive. You can use bash [string manipulation](https://tldp.org/LDP/abs/html/string-manipulation.html) to achieve this:

`DRY_RUN="${DRY_RUN//[-_]/}"` strips out dashes and underscores
`DRY_RUN="${DRY_RUN^^}"` converts to uppercase

And since only exactly one argument makes sense and otherwise it's probably an error, you can also error out if someone passes any other argument (after converting)

```bash
if [[ -z "${DRY_RUN}" ]] && [[ "${DRY_RUN}" != "DRYRUN" ]]; then
  echo "Unexpected argument '$2'. Should be DRYRUN (case and punctuation insensitive) if present"
  exit 1
fi
```

Left all this as a suggested edit above (where it would actually be happening. Also, nit, always quote and brace-delimit variables: https://google.github.io/styleguide/shellguide.html#s5.7-quoting

```suggestion
if [[ "${DRY_RUN}" == "DRYRUN" ]]; then
```